Hunter 0.5.3

tatraian opened the pull request:
Huge amount of changes in CMakeLists.txt:
- boost comes from hunter
- it is able to create both static and dynamic libraries
- install has been updated to create package description

tatraian committed:
Package hunterized

tatraian committed:
hunter updated to v0.16.30

ruslo commented on CMakeLists.txt:
```diff
@@ -254,87 +253,109 @@ else()
 endif()
 
-set(INCLUDE_INSTALL_ROOT_DIR include)
+set(INCLUDE_INSTALL_ROOT_DIR ${CMAKE_INSTALL_PREFIX}/include)
```
Don't do this, see http://cgold.readthedocs.io/en/latest/tutorials/install/install-prefix.html

ruslo commented on CMakeLists.txt:
```diff
@@ -254,87 +253,109 @@ else()
-  COMPILE_FLAGS "${yaml_c_flags} ${yaml_cxx_flags}"
+set(target_to_export yaml-cpp-static)
+
+add_library(yaml-cpp-static STATIC ${library_sources})
```
Library definition without `STATIC` is a correct approach, see http://cgold.readthedocs.io/en/latest/tutorials/libraries/static-shared.html

tatraian committed:
Removed duplicated target for library building.

tatraian commented:
Hello,

I removed the duplicated target creation, so now you can set the library type by `BUILD_SHARED_LIB`. Otherwise it contains the original file. And I added CMake target exported file to install.
Do I remove pkgconfig files?

ruslo commented on CMakeLists.txt:
```diff
@@ -13,13 +13,22 @@ if(POLICY CMP0015)
-project(YAML_CPP)
+include(CheckCXXCompilerFlag)
+hunter_add_package(Boost)
+find_package(Boost CONFIG REQUIRED)
```
Where do we use found Boost libraries?

tatraian replied:
Sorry. I wrote the `find_package` in the code here. The original file also contains it at line 125. And after it adds Boost include dir to the include directories. I erase it from here.

ruslo commented:
> Do I remove pkgconfig files?

If they doesn't trigger any errors in Hunter we can leave it as is. I think making minimal changes compared to upstream is a good approach.

tatraian committed:
Remove duplicated find_package(boost)

ruslo commented on CMakeLists.txt:
```diff
@@ -113,7 +119,7 @@ endif()
 
-find_package(Boost REQUIRED)
+find_package(Boost CONFIG REQUIRED)
 include_directories(${Boost_INCLUDE_DIRS})
```
I expect `Boost::boost` somewhere instead.

ruslo reviewed (changes requested):
Try to install such configuration. There are some variables with full paths in installed CMake configs. This makes the package unrelocatable. Grep for YAML_CPP_INCLUDE_DIR and IMPORTED_LOCATION_DEBUG after install done.

tatraian committed:
boost include comes from target_link_libraries

tatraian committed:
refactored install

tatraian commented:
Sorry, I didn't find what is wrong so I just simply rewrote the install part... But I think now it works.

ruslo commented on CMakeLists.txt:
```diff
@@ -272,7 +277,8 @@ set(_INSTALL_DESTINATIONS
   COMPILE_FLAGS "${yaml_c_flags} ${yaml_cxx_flags}"
-)
+  )
+target_link_libraries(yaml-cpp Boost::boost)
```
I think it should be `target_link_libraries(yaml-cpp PUBLIC Boost::boost)`

ruslo commented on CMakeLists.txt:
```diff
@@ -298,24 +304,48 @@ if(MSVC)
-	DESTINATION ${INCLUDE_INSTALL_DIR}
-	FILES_MATCHING PATTERN "*.h"
-)
+set(VERSION ${YAML_CPP_VERSION})
```
As far as I understand the `VERSION` variable used only in `write_basic_package_version_file`. So it can be just `write_basic_package_version_file(... VERSION "${YAML_CPP_VERSION}")`.

tatraian committed:
VERSION settings eliminated and Boost include becomes PUBLIC

tatraian commented:
I fixed them.

ruslo commented:
There should be a `find_dependency(Boost CONFIG)` call in config.cmake.in template. Also PACKAGE_INIT and check_required_components, see https://github.com/forexample/package-example/blob/master/Foo/cmake/Config.cmake.in

tatraian committed:
rewritten config.cmake template

tatraian commented:
I rewrote the config template. I proposed that you want to write `find_package` and not `find_dependecies`.

ruslo commented:
> I proposed that you want to write find_package and not find_dependecies.

It should be `find_dependency` if you're planning to be pedantic :) `find_package` will do the trick for us too.
